From: Gunnar Wrobel <p@rdus.de>
Subject: [PATCH] t/dimp/H/BL/CloseContextMenu

Close the context menu when clicking into the Dimp menu.

Provided by Boris Lorbeer <B.Lorbeer@science-computing.de>

Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 horde-webmail/dimp/config/menu.php |    2 +-
 horde-webmail/dimp/js/DimpBase.js  |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/horde-webmail/dimp/config/menu.php b/horde-webmail/dimp/config/menu.php
index b5e0a5c..a079da6 100644
--- a/horde-webmail/dimp/config/menu.php
+++ b/horde-webmail/dimp/config/menu.php
@@ -31,7 +31,7 @@
 
 $site_menu = array(
     'folders' => array(
-        'action' => 'DimpBase.go("app:horde", "' . Horde::url($GLOBALS['registry']->get('webroot', 'imp') . '/folders.php', true) . '")',
+        'action' => 'DimpCore.DMenu.close(); DimpBase.go("app:horde", "' . Horde::url($GLOBALS['registry']->get('webroot', 'imp') . '/folders.php', true) . '")',
         'text' => _('Folder Subscription'),
         'icon' => $GLOBALS['registry']->getImageDir('horde') . '/prefs.png'),
 );
diff --git a/horde-webmail/dimp/js/DimpBase.js b/horde-webmail/dimp/js/DimpBase.js
index c4eb9f4..d90f09b 100644
--- a/horde-webmail/dimp/js/DimpBase.js
+++ b/horde-webmail/dimp/js/DimpBase.js
@@ -1507,6 +1507,7 @@ var DimpBase = {
 
                 case 'folder':
                 case 'special':
+                    DimpCore.DMenu.close();
                     e.stop();
                     return this.go('folder:' + li.readAttribute('mbox'));
                     break;
@@ -1903,7 +1904,7 @@ var DimpBase = {
         tmp = $('applicationfolders');
         if (tmp) {
             tmp.select('ul li.custom a').each(function(s) {
-                C({ d: s, f: this.go.bind(this, 'app:' + s.readAttribute('app')) });
+                C({ d: s, f: function(data){DimpCore.DMenu.close(); this.go('app:' + s.readAttribute('app'), data)}.bind(this) });
             }, this);
         }
 
-- 
tg: (08b89e6..) t/dimp/H/BL/CloseContextMenu (depends on: t/kronolith/HK/GW/AnonymousiCalDownload)
-- 
TOPGIT patch commit log
=======================

commit b4b3190a4fb88e1843ff86adde4897dea21046db
Author: Gunnar Wrobel <p@rdus.de>
Date:   Mon Jul 20 09:39:24 2009 +0200

    Close the context menu when clicking into the Dimp menu.
